### DISCUSSION TITLE
Ledger Live app store

### DISCUSSION THREAD

**Discussion by aido** [aido/app-seed-tool#15] [General]
Ledger may be considering adding this app to their Ledger Live “My Ledger“ app list

https://github.com/LedgerHQ/ledger-app-database/pull/103#issuecomment-1761191937

**Comment by aido** [comment]
Ledger Seed Tool application moved to the "Pending functional review" column of Ledger's [Integration Board](https://airtable.com/appIHCUneslDfKjXu/shrqrpH9y9VRorS4J/tblw7wsOQj5Fb2UzA).

**Comment by aido** [comment]
Seed Tool app needs to be audited before being made available in Ledger Live app store.

https://developers.ledger.com/docs/device-app/submission/security-audit

Ledger are trying to find a way to take charge of the audit. They have a very limited bandwidth though, so need to find a slot

**Reply by clouedoc** [reply]
Rooting for you! SSS is something they should have programmed themselves

**Reply by aido** [reply]
> SSS is something they should have programmed themselves

I agree.

But https://github.com/LedgerHQ/ledger-nano-s/issues/54 has been open for nearly 5 years now with no progress, so Ledger do not seem to be in any hurry with offering SSS on their devices. Also, if Ledger had written an SSS app I fear it may have been their own implementation that would not be interoperable like SSKR is.

**Comment by aido** [comment]
> Ledger may be considering adding this app to their Ledger Live “My Ledger“ app list
> 
> [LedgerHQ/ledger-app-database#103 (comment)](https://github.com/LedgerHQ/ledger-app-database/pull/103#issuecomment-1761191937)

Seed Tool app now added to Ledger App database

**Comment by aido** [comment]
Ledger Seed Tool application moved to the "To be released" column of Ledger's [Integration Board](https://airtable.com/appIHCUneslDfKjXu/shrqrpH9y9VRorS4J/tblw7wsOQj5Fb2UzA).

**Comment by aido** [comment]
Ledger's security review of app-seed-tool is complete!

A Ledger blogpost has been written but must first be validated by Ledger's head of communication before being published. Then the app will be ready to be released.

**Reply by clouedoc** [reply]
Finally!

**Reply by DmKoshelek** [reply]
Hi @aido, do you have any updates about the release date? I am really looking forward for your extension. Thanks!

**Reply by aido** [reply]
Hi @DmKoshelek,

Two weeks ago I pinged Ledger (Victor F) on Ledger's Discord channel to ask what the status was. Victor told me that he was still waiting on the blog post to be validated by the head of communication but he hoped the app would be released early the following week.  That didn't happen :disappointed: so I pinged Victor again last week to be told that the blog post was still pending approval.

Maybe @lpascal-ledger or @xchapron-ledger might know the release status as they have been updating the Ledger fork of the app?

**Reply by lpascal-ledger** [reply]
Unfortunately we don't have much visibility on this kind of topic. I'll try to get some.
In any case, your app was deployed and is available on provider 4 (which is our testing provider). It is not a proper release, but your most enthusiastic users could already test the app right now.

**Reply by DmKoshelek** [reply]
@lpascal-ledger Thanks for the advice, I changed provide and now I can finally use the app!

**Reply by lpascal-ledger** [reply]
To be clear, as I said provider 4 is a **testing** provider where apps are released a bit sooner, but they can come with bug or security issues, so stay cautious with your usage, you're on your own :)

**Reply by aido** [reply]
Thanks for the tip @lpascal-ledger 

I just installed the app from provider 4.
Wow! It is only 40KB in size on a Nano S.
Previously, when building and testing the app I thought the binary was much much bigger. This means that a future version may be able to fit some BIP85 functionality onto Nano S devices. We'll see, no promises, adding BIP85 menus may make the UI too messy.
But at least for now I know that space is not the limitation I previously thought it was.

**Reply by aido** [reply]
To add, for now BIP85 functionality is lower down my list of priorities. The current priority is getting a bug free version of the app working on Stax. See LedgerHQ/app-seed-tool#5.

It would be great if Ledger provided me with a Stax device for testing!  :grin:

**Reply by aido** [reply]
@DmKoshelek 

May I suggest that once you have app-seed-tool installed from Ledger Live that you change the settings back to the original so you are NOT using provider 4 anymore.
Otherwise you run the risk of accidentally installing unstable versions of other apps without being aware.

**Comment by aido** [comment]
[v1.7.4](https://github.com/aido/app-seed-tool/releases/tag/v1.7.4) will most likely be the stable release that will be available to install from Ledger Live.

This version is being sent to Ledger Donjon for approval and then hopefully will be made available on Ledger Live. A Ledger blog post for app-seed-tool has also been approved so now we just wait for a 👍 from Donjon.

**Reply by aido** [reply]
[v1.7.4](https://github.com/aido/app-seed-tool/releases/tag/v1.7.4) is planned to be audited in the sprint starting Monday 2024-07-08.

**Reply by aido** [reply]
[v1.7.4](https://github.com/aido/app-seed-tool/releases/tag/v1.7.4) has passed the Ledger security audit so brace yourselves, official release on production is imminent.

**Comment by aido** [comment]
Finally!!!!

The Ledger Seed Tool app has been released and [v1.7.4](https://github.com/aido/app-seed-tool/releases/tag/v1.7.4) can be installed from Ledger Live.

The app is available for all Ledger Nano devices; Nano S, Nano S Plus and Nano X.
The Ledger Stax version is still a work in progress and the next version of the app will include all the screens needed to generate Shamir Secret Shares on Stax too.

If you would like to see some screenshots or a short animation of the various menus then they are available here:
[Screenshots](https://github.com/aido/app-seed-tool/tree/screenshots/tests/functional/screenshots)

A blogpost will soon be released on the Ledger website alongside a Ledger Help Centre article redirecting to it and mentioning that Stax support is still a work in progress

**Reply by aido** [reply]
The Ledger blog post on app-seed-tool is now available at https://www.ledger.com/blog/seed-tool-app

**Comment by aido** [comment]
The Ledger Seed Tool application [v1.8.0](https://github.com/aido/app-seed-tool/releases/tag/v1.8.0) is now available to install and test from provider 4 on Ledger Live. This is a **testing** provider, so please be aware of potential caveats.

See:
- https://github.com/aido/app-seed-tool/discussions/15#discussioncomment-9303759
- https://github.com/aido/app-seed-tool/discussions/15#discussioncomment-9306764


With v1.8.0, the application now supports Stax and Flex devices. The larger screens on these devices provide a significant improvement over the smaller and more limited Nano series.
Entering long phrases on Nano devices can be tedious, but the touch-sensitive keyboards on Stax and Flex make input much faster and easier.

> [!NOTE]
> Nano S devices have been end-of-life for over a year, so v1.8.0 will not be deployed for them. However, [v1.7.4](https://github.com/aido/app-seed-tool/releases/tag/v1.7.4) remains available for installation via Ledger Live.
